staging/src/k8s.io/dynamic-resource-allocation/resourceslice/resourceslicecontroller.go

```diff
@@ -222,7 +313,7 @@ func (c *Controller) initInformer(ctx context.Context) error {
 	}, func(options *metav1.ListOptions) {
 		options.FieldSelector = selector.String()
 	})
-	c.sliceStore = informer.GetIndexer()
+	c.sliceStore = cache.NewIntegerResourceVersionMutationCache(informer.GetStore(), informer.GetIndexer(), c.mutationCacheTTL, true /* includeAdds */)
```

Comment:
Another improvement idea: in the event handlers below, add pools with a delay (`AddAfter` instead of `Add`). This doesn't matter anymore when we add slices because of the mutation cache, but if we delete slices, it might reduce the number of redundant delete calls because the sync should run after the informer cache is up-to-date again and thus all executed deletes are known.

Reply:
Implemented.

Reply:
Before:
```
  I1029 15:00:33.588378  313236 resourceslicecontroller.go:693] "Deleting obsolete resource slice" poolName="network-attached" slice="dra-9321-zlxmg" deleteOptions={"preconditions":{"uid":"b54363dd-4cc7-4518-83e5-caae96907e8d","resourceVersion":"929"}}
  I1029 15:00:33.596386  313236 resourceslicecontroller.go:366] "ResourceSlice delete" slice="dra-9321-zkch8"
  I1029 15:00:33.637719  313236 resourceslicecontroller.go:489] "Existing slices" poolName="network-attached" obsolete=["dra-9321-zlxmg"] current=["dra-9321-pbssq"]
  I1029 15:00:33.637777  313236 resourceslicecontroller.go:523] "Matched existing slice" poolName="network-attached" slice="dra-9321-pbssq" matchIndex=0
  I1029 15:00:33.637831  313236 resourceslicecontroller.go:561] "Completed comparison" poolName="network-attached" numObsolete=1 numMatchedSlices=1 numChangedMatchedSlices=0 numNewSlices=0
  I1029 15:00:33.637862  313236 resourceslicecontroller.go:577] "Kept generation because at most one update API call is necessary" poolName="network-attached" generation=2
  I1029 15:00:33.637900  313236 resourceslicecontroller.go:693] "Deleting obsolete resource slice" poolName="network-attached" slice="dra-9321-zlxmg" deleteOptions={"preconditions":{"uid":"b54363dd-4cc7-4518-83e5-caae96907e8d","resourceVersion":"929"}}
  I1029 15:00:33.645496  313236 resourceslicecontroller.go:366] "ResourceSlice delete" slice="dra-9321-zlxmg"
  I1029 15:00:33.666990  313236 resourceslicecontroller.go:699] "Resource slice was already deleted earlier" poolName="network-attached" slice="dra-9321-zlxmg"
```

After:
```
  I1029 15:07:47.792780  321917 resourceslicecontroller.go:561] "Completed comparison" poolName="network-attached" numObsolete=100 numMatchedSlices=0 numChangedMatchedSlices=0 numNewSlices=1
  I1029 15:07:47.792866  321917 resourceslicecontroller.go:581] "Bumped generation by one" poolName="network-attached" generation=2
  I1029 15:07:47.792911  321917 resourceslicecontroller.go:657] "Creating new resource slice" poolName="network-attached"
  I1029 15:07:47.795519  321917 resourceslicecontroller.go:338] "ResourceSlice add" slice="dra-8280-vh92n"
...
  I1029 15:07:50.476812  321917 resourceslicecontroller.go:366] "ResourceSlice delete" slice="dra-8280-zg8w8"
  I1029 15:07:57.796962  321917 resourceslicecontroller.go:489] "Existing slices" poolName="network-attached" obsolete=[] current=["dra-8280-vh92n"]
```

There's only a ~7 second delay between that last informer event and the next sync. That's because the workqueue implementation only shortens the delay for an already pending item, but does not lengthen it. Therefore our 30 second delay counts down from first add event at `15:07:47.795519`.

 We would like to reset the timer for each incoming event, but it's still better than no delay.